### ISSUE TITLE
migrate the membership DB to LDAP

### ISSUE THREAD

**Issue by dckc** [rchain/bounties#513]
context: [today's meeting](https://github.com/rchain/bounties/issues/446#issuecomment-373931669)

_This is assigned to @ian-bloom only as a proxy for @jeremybusk , who doesn't seem to be connected to this repo in the relevant way._

cc @nashef

**Comment by dckc** [comment]
@ian-bloom it occurs to me that in order to complete this, we need to update the code / service behind https://member.rchain.coop/ .

 - Do you expect @jeremybusk to do that?
 - Do you want me to look into it? If so, I need either
   - access to the current source code or
   - encouragement to abandon the current code and start over
     - my approach would be to design it as a Rholang app and use [Capper][] (an ocap node.js library) as an approximation to start with.

[Capper]: https://github.com/dckc/Capper

**Comment by dckc** [comment]
I'm a lot more comfortable with version-controlled text formats for critical data.

I suppose there's a reasonably straightforward way to dump the LDAP database and check it in somewhere...

**Comment by dckc** [comment]
In #428 I'm reminded you prefer multi-language wordpress for https://member.rchain.coop/ . wordpress and PHP are hopeless when it comes to object capability security.

While I have experience with Capper, I get the impression the current code is written in scala. It's at least moderately straightforward to maintain object capability discipline in scala. Plus, more people learning scala means more people ready to contribute to the platform.

**Comment by kitblake** [comment]
+1 Yes, let's **please** migrate the member database into LDAP. That's exactly what we need to scale.

**Comment by dckc** [comment]
@kitblake care to elaborate? On reflection, I don't understand exactly what motivates using LDAP over an SQL database such as the one built in to WordPress or drupal.

In particular: I'm increasingly getting the impression that member profiles will be edited using WordPress. Does WordPress know how to read / write LDAP?

**Comment by dckc** [comment]
Yup... this WP plugin looks pretty mature and offers updating LDAP attributes via the WP UI and such...

[Active Directory Integration / LDAP Integration](https://wordpress.org/plugins/ldap-login-for-intranet-sites/)
